src/fides/api/task/conditional_dependencies/privacy_request/privacy_request_data.py

```diff
@@ -0,0 +1,124 @@
+from fides.api.schemas.policy import Policy as PolicySchema
+from fides.api.task.conditional_dependencies.privacy_request.convenience_fields import (
+    get_policy_convenience_fields,
+)
+from fides.api.task.conditional_dependencies.util import (
+    extract_nested_field_value,
+    set_nested_value,
+)
+
+
+class PrivacyRequestDataTransformer:
+    """
+    Transforms a PrivacyRequest ORM object into a dictionary structure for condition evaluation.
+
+    This class extracts the fields referenced in conditions and builds a minimal nested structure
+    for evaluation. Privacy request fields must be prefixed with the "privacy_request" namespace.
+
+    Note: Privacy request relationships are eager loaded in the execute_request_tasks.py file.
+    """
+
+    def __init__(self, privacy_request: PrivacyRequest):
+        self.privacy_request = privacy_request
+        self.policy_data = self._transform_policy_object(privacy_request.policy)
+        self.identity_data = self.privacy_request.get_persisted_identity()
+        self.custom_privacy_request_fields_data = (
+            self.privacy_request.get_persisted_custom_privacy_request_fields()
+        )
+
+    def to_evaluation_data(self, field_addresses: set[str]) -> dict[str, Any]:
+        """
+        Transforms the PrivacyRequest into an evaluation-ready dictionary structure.
+
+        Args:
+            field_addresses: Set of field addresses referenced in conditions
+        """
+        if not field_addresses:
+            return {}
+
+        result: dict[str, Any] = {"privacy_request": {}}
+        field_paths: set[str] = {addr.replace(":", ".") for addr in field_addresses}
+
+        # Extract each field dynamically based on its path
+        for field_path in field_paths:
+            value = self._extract_value_by_path(field_path)
+            privacy_request_data: Optional[dict[str, Any]] = set_nested_value(
+                field_path.split("."), value
+            )
+            if privacy_request_data:
+                result = deep_update(result, privacy_request_data)
+
+        return result
+
+    def _extract_value_by_path(self, field_address: str) -> Any:
+        """
+        Extracts a value from PrivacyRequest by following the field address path.
+
+        Args:
+            field_address: The field address path (e.g., "privacy_request.location", "privacy_request.policy.rules")
+
+        """
+        # first part will always be privacy_request
+        parts: list[str] = field_address.split(".")[1:]
+        current: Any = self.privacy_request
+
+        # Track the identity object if we're extracting from identity
+        if parts[0] == "policy":
+            current = self.policy_data
+        elif parts[0] == "identity":
+            current = self.identity_data
+        elif parts[0] == "custom_privacy_request_fields":
+            current = self.custom_privacy_request_fields_data
+
+        if current != self.privacy_request:
```

Comment:
Is this because the default is "anything in the privacy request doesn't have an explicit start path"? What if I pass in `privacy_request.xxx.yyy`?